psycopg3/psycopg3/cursor.py

```diff
@@ -364,6 +373,7 @@ def _execute_results(self, results: Sequence["PGresult"]) -> None:
 
         self._results = list(results)
         self.pgresult = results[0]
+        self._make_row = self._row_factory(self)
```

Comment:
Need a test to verify it works ok and doesn't regress for queries returning multiple results.

Reply:
Done in commit 'Add tests for cursor's row_factory with nextset()'.